tidb.go

```diff
@@ -292,4 +277,74 @@ func DestoryTiDBServer(t *TiDBServer) {
 	t.closeDomainAndStorage()
 	t.CloseGracefully()
 	tidbServer = nil
+}
+
+func RemoveTiDBServerVar(){
+	tidbServer = nil
+}
+
+/*
+ * setDBInfoMeta / setTableInfoMeta is used to store the correct dbInfo and tableInfo into
+ * TiDB-lite meta layer directly. Cause the dbInfo and tableInfo is extracted from ddl history
+ * job, so it's correctness is guaranteed.
+ */
+
+func (t *TiDBServer) SetDBInfoMeta(newDBs []*model.DBInfo) error {
+	err := kv.RunInNewTxn(t.storage, true, func(txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		var err1 error
+		originDBs, err1 := t.ListDatabases()
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		// delete the origin db with same ID and name.
+		deleteDBIfExist := func(newDB *model.DBInfo) error {
+			for _, originDB := range originDBs {
+				if originDB.ID == newDB.ID {
+					if err1 = t.DropDatabase(originDB.ID); err1 != nil {
+						return errors.Trace(err1)
+					}
+				}
+			}
+			return nil
+		}
+
+		// store meta in kv storage.
+		for _, newDB := range newDBs {
+			if err1 = deleteDBIfExist(newDB); err1 != nil {
+				return errors.Trace(err1)
+			}
+			// create database.
+			if err1 = t.CreateDatabase(newDB); err1 != nil {
+				return errors.Trace(err1)
+			}
+			// create table.
+			for _, newTable := range newDB.Tables {
+				// like create table do, it should rebase to AutoIncID-1.
+				if err1 = t.CreateTableAndSetAutoID(newDB.ID, newTable, newTable.AutoIncID-1); err1 != nil {
+					return errors.Trace(err1)
+				}
+			}
+		}
+		/*
+		 * update schema version here, when it exceed 100, domain reload will fetch all tables from meta directly
+		 * rather than applying schemaDiff one by one.
+		 */
+		for i:=0; i<=105; i++ {
+			_, err := t.GenSchemaVersion()
+			if err != nil {
+				return errors.Trace(err)
+			}
+		}
+		return nil
+	})
+	return err
```

Comment:
We can put domain.Reload in this function

Reply:
Addressed.